bids/variables/variables.py

```diff
@@ -408,7 +408,9 @@ def to_dense(self, sampling_rate=None):
         else:
             sampling_rate = bin_sr
 
-        duration = int(math.ceil(bin_sr * self.get_duration()))
+        duration = math.ceil(  # Round up to nearest second
+            round(bin_sr * self.get_duration(), 3)  # Cut off at millisecond precision
+        )
```

Comment:
Python 3.11:

```python
>>> sum([480.1] * 9)
4390.9
```

Python 3.12:

```python
>>> sum([480.1] * 9)
4320.900000000001
```

It seems to me that we don't care about errors in the sub-millisecond range, so round to the nearest millisecond and then round up to the nearest second. Reduces the scope of potential bad rounding to `N.4995`.